src/libraries/System.Private.CoreLib/src/System/IO/FileStatus.SetTimes.OSX.cs

```diff
@@ -19,23 +26,23 @@ internal void SetCreationTime(string path, DateTimeOffset time, bool asDirectory
             // great care.
             long seconds = time.ToUnixTimeSeconds();
             long nanoseconds = UnixTimeSecondsToNanoseconds(time, seconds);
-            Interop.Error error = SetCreationTimeCore(path, seconds, nanoseconds);
+            Interop.Error error = SetCreationTimeCore(handle, path, seconds, nanoseconds);
 
             if (error == Interop.Error.SUCCESS)
             {
                 InvalidateCaches();
             }
             else if (error == Interop.Error.ENOTSUP)
             {
-                SetAccessOrWriteTimeCore(path, time, isAccessTime: false, checkCreationTime: false, asDirectory);
+                SetAccessOrWriteTimeCore(handle, path, time, isAccessTime: false, checkCreationTime: false, asDirectory);
             }
             else
             {
                 Interop.CheckIo(error, path, asDirectory);
             }
         }
 
-        private static unsafe Interop.Error SetCreationTimeCore(string path, long seconds, long nanoseconds)
+        private static unsafe Interop.Error SetCreationTimeCore(SafeFileHandle? handle, string? path, long seconds, long nanoseconds)
```

Comment:
This should assert that one and only one of the handle/path inputs is null. Same for any other such leaf functions.